server/src/main/java/org/elasticsearch/common/geo/TriangleTreeReader.java

```diff
@@ -58,8 +58,7 @@ public void reset(BytesRef bytesRef) throws IOException {
      */
     public Extent getExtent() {
         if (treeOffset == 0) {
-            // TODO: Compress serialization of extent
-            input.setPosition(extentOffset);
+            getDimensionalShapeType(); // skip DimensionalShapeType
```

Comment:
I will add the dimensional type as the first byte of the serialise tree and skip it together with the centroid.

Reply:
I left this because of the variable encoding of the enum. in order to properly skip it, it must first be read and cached. is that preferred? or is changing the encoding to be a byte better? since it is known to be a small amount of constants, it could be serialized to a byte with tests making sure any future changes do not break this

Reply:
I would expect the enum to be one byte, ordinal from 1 to 7 must fit in a byte

Reply:
I agree, but I'd be happier to be explicit about this, instead of just trusting that vint will return a byte. I'll update!